File: android/framework_old/src/main/java/com/dreampany/nearby/model/Peer.java
package com.dreampany.nearby.model;

import com.google.common.primitives.Longs;

/**
 * Created by Hawladar Roman on 6/4/2018.
 * BJIT Group
 * hawladar.roman@bjitgroup.com
 */
public class Peer {

    public enum State {
        LIVE, DEAD
    }

    private final long id;
    private byte[] data;

    public Peer(long id) {
        this(id, null);
    }

    public Peer(long id, byte[] data) {
        this.id = id;
        this.data = data;
    }

    @Override
    public boolean equals(Object inObject) {
        if (this == inObject) return true;
        if (inObject == null || getClass() != inObject.getClass()) return false;

        Peer peer = (Peer) inObject;
        if (id != peer.id) return false;
        return true;
    }

    @Override
    public int hashCode() {
        return Longs.hashCode(id);
    }

    @Override
    public String toString() {
        return "peer: \n" + "id - " + id + "\n" +
                "data - " + (data == null ? "null" : new String(data));
    }

    public void setData(byte[] data) {
        this.data = data;
    }

    public long getId() {
        return id;
    }

    public byte[] getData() {
        return data;
    }
}

File: android/network/src/main/java/com/dreampany/network/nearby/data/model/Peer.kt
package com.dreampany.network.nearby.data.model

import com.dreampany.network.misc.Constant
import com.google.common.base.Objects

/**
 * Created by roman on 7/2/21
 * Copyright (c) 2021 butler. All rights reserved.
 * ifte.net@gmail.com
 * Last modified $file.lastModified
 */
data class Peer(
    val id: String,
    var raw: ByteArray = Constant.Default.BYTEARRAY
) {

    enum class State { LIVE, DEAD }

    override fun hashCode(): Int = Objects.hashCode(id)

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val item = other as Peer
        return Objects.equal(this.id, item.id)
    }

    override fun toString(): String = "Peer[id:$id]"
}